deeppavlov/models/ner/network_ontonotes.py

```diff
@@ -0,0 +1,187 @@
+                                                                  sequence_lengths)
+
+        # Initialize session
+        if sess is None:
+            sess = tf.Session()
+
+        self._ner_tagger = SennaNERTagger('download/senna/')
+        self._pos_tagger = SennaChunkTagger('download/senna/')
+
+        self._x_w = x_word
+        self._x_c = x_char
+        self._x_capi = x_capi
+        self.x_pos = x_pos
+        self.x_ner = x_ner
+        self._y_true = y_true
+        self._mask = mask
+        self._sequence_lengths = sequence_lengths
+        self._token_embeddings_dim = token_embeddings_dim
+
+        self._pos_dict = pos_vocab
+        self._ner_dict = ner_vocab
+        self._tag_dict = tag_vocab
+
+        self._logits = logits
+        self._trainsition_params = trainsition_params
+
+        self._sess = sess
+        sess.run(tf.global_variables_initializer())
+        self._embedder = embedder
+
+    def load(self, model_file_path):
+        saver = tf.train.Saver(tf.trainable_variables())
+        saver.restore(self._sess, model_file_path)
+
+    @staticmethod
+    def to_one_hot(x, n):
+        b = np.zeros([len(x), n], dtype=np.float32)
+        for q, tok in enumerate(x):
+            b[q, tok] = 1
+        return b
+
+    def tokens_batch_to_numpy_batch(self, batch_x):
+        """ Convert a batch of tokens to numpy arrays of features"""
+        x = dict()
+        batch_size = len(batch_x)
+        max_utt_len = max([len(utt) for utt in batch_x])
+
+        # Embeddings
+        x['emb'] = self._embedder(batch_x)
+
+        # Capitalization
+        x['capitalization'] = np.zeros([batch_size, max_utt_len], dtype=np.float32)
+        for n, utt in enumerate(batch_x):
+            x['capitalization'][n, :len(utt)] = [tok[0].isupper() for tok in utt]
+
+        # POS
+        n_pos = len(self._pos_dict)
+        x['pos'] = np.zeros([batch_size, max_utt_len, n_pos])
+        for n, utt in enumerate(batch_x):
+            token_tag_pairs = self._pos_tagger.tag(utt)
+            pos_tags = list(zip(*token_tag_pairs))[1]
+            pos = np.array([self._pos_dict[p] for p in pos_tags])
+            pos = self.to_one_hot(pos, n_pos)
+            x['pos'][n, :len(pos)] = pos
+
+        # NER
+        n_ner = len(self._ner_dict)
+        x['ner'] = np.zeros([batch_size, max_utt_len, n_ner])
+        for n, utt in enumerate(batch_x):
+            token_tag_pairs = self._ner_tagger.tag(utt)
+            ner_tags = list(zip(*token_tag_pairs))[1]
+            ner = np.array([self._ner_dict[p] for p in ner_tags])
+            ner = self.to_one_hot(ner, n_ner)
+            x['ner'][n, :len(ner)] = ner
+
+        # Mask for paddings
+        x['mask'] = np.zeros([batch_size, max_utt_len], dtype=np.float32)
+        for n in range(batch_size):
+            x['mask'][n, :len(batch_x[n])] = 1
+
+        return x
+
+    def train_on_batch(self, x_word, x_char, y_tag):
+        raise NotImplementedError
```

Comment:
If it cannot be trained it could just be a serializable and not an NNModel. And a config for it could not have an `in_y` field at all